sdk/core/azure-core-amqp/src/common/global_state.cpp

```diff
@@ -8,17 +8,24 @@
 
 #include <azure_c_shared_utility/platform.h>
 
-namespace Azure { namespace Core { namespace _internal { namespace Amqp { namespace Common {
-  namespace _detail {
+namespace Azure { namespace Core { namespace Amqp { namespace Common { namespace _detail {
 
-    GlobalState::GlobalState()
+  GlobalState::GlobalState()
+  {
+    if (platform_init())
     {
-      if (platform_init())
-      {
-        throw std::runtime_error("Could not initialize platform.");
-      }
+      throw std::runtime_error("Could not initialize platform.");
     }
+  }
 
-    GlobalState::~GlobalState() { platform_deinit(); }
+  GlobalState::~GlobalState() { platform_deinit(); }
```

Comment:
This will never be called because no one is going to delete the global state (and if we expect customers/partners to call delete on it, we can't use placement new; probably the right thing to do then is to override operator delete).

Reply:
At some point in the future we might find a need for this destructor, it won't hurt to have it.